fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java

```diff
@@ -68,35 +93,142 @@ public List<String> listPartitionNames(String databaseName, String tableName) {
+            databases.put(dbName, db);
+            return db;
+        } else {
+            LOG.error("Paimon database {}.{} done not exist.", catalogName, dbName);
+            return null;
+        }
     }
 
     @Override
     public Table getTable(String dbName, String tblName) {
         Identifier identifier = new Identifier(dbName, tblName);
-        org.apache.paimon.table.Table paimonTable;
+        if (tables.containsKey(identifier)) {
+            return tables.get(identifier);
+        }
+        org.apache.paimon.table.Table paimonNativeTable;
         try {
-            paimonTable = this.paimonCatalog.getTable(identifier);
+            paimonNativeTable = this.paimonNativeCatalog.getTable(identifier);
         } catch (Catalog.TableNotExistException e) {
-            throw new StarRocksConnectorException("Paimon table {}.{} does not exist.", dbName, tblName);
+            LOG.error("Paimon table {}.{} does not exist.", dbName, tblName);
+            return null;
         }
-        List<DataField> fields = paimonTable.rowType().getFields();
+        List<DataField> fields = paimonNativeTable.rowType().getFields();
         ArrayList<Column> fullSchema = new ArrayList<>(fields.size());
         for (DataField field : fields) {
             String fieldName = field.name();
             Type fieldType = ColumnTypeConverter.fromPaimonType(field.type());
             Column column = new Column(fieldName, fieldType, true);
             fullSchema.add(column);
         }
-        return new PaimonTable(catalogName, dbName, tblName, fullSchema, catalogType, metastoreUris, warehousePath, paimonTable);
+        PaimonTable table = new PaimonTable(catalogName, dbName, tblName, fullSchema,
+                catalogType, metastoreUris, warehousePath, paimonNativeTable);
+        tables.put(identifier, table);
+        return table;
+    }
+
+    @Override
+    public List<RemoteFileInfo> getRemoteFileInfos(Table table, List<PartitionKey> partitionKeys,
+                                                   long snapshotId, ScalarOperator predicate, List<String> fieldNames) {
+        RemoteFileInfo remoteFileInfo = new RemoteFileInfo();
+        PaimonTable paimonTable = (PaimonTable) table;
+        PaimonFilter filter = new PaimonFilter(paimonTable.getDbName(), paimonTable.getTableName(), predicate, fieldNames);
+        if (!paimonSplits.containsKey(filter)) {
+            ReadBuilder readBuilder = paimonTable.getNativeTable().newReadBuilder();
+            int[] projected = fieldNames.stream().mapToInt(name -> (paimonTable.getFieldNames().indexOf(name))).toArray();
+            List<Predicate> predicates = extractPredicates(paimonTable, predicate);
+            List<Split> splits = readBuilder.withFilter(predicates).withProjection(projected).newScan().plan().splits();
+            PaimonSplitsInfo paimonSplitsInfo = new PaimonSplitsInfo(predicates, splits);
+            paimonSplits.put(filter, paimonSplitsInfo);
```

Comment:
why attach filter with split info?

Reply:
The paimon reader api can directly use the columns to be read and the filter conditions to return the corresponding splits, that is, both column pruning and predicate pushdown are completed during the process of obtaining paimon splits